src/core/js/models/lockingModel.js

```diff
@@ -1,138 +1,134 @@
-      if (Object.keys(this._lockedAttributes).length === 0) {
-        delete this._lockedAttributes;
-        delete this._lockedAttributesValues;
-      }
-    },
-
-    isLocking: function(attrName) {
-      const isCheckingGeneralLockingState = (attrName === undefined);
-      const isUsingLockedAttributes = Boolean(this.lockedAttributes || this._lockedAttributes);
-
-      if (isCheckingGeneralLockingState) {
-        return isUsingLockedAttributes;
-      }
-
-      if (!isUsingLockedAttributes) return false;
-
-      if (!this._lockedAttributes) {
-        this._lockedAttributes = _.result(this, 'lockedAttributes');
-      }
-
-      const isAttributeALockingAttribute = this._lockedAttributes.hasOwnProperty(attrName);
-      if (!isAttributeALockingAttribute) return false;
-
-      if (!this._lockedAttributesValues) {
-        this._lockedAttributesValues = {};
-      }
-
-      if (!this._lockedAttributesValues[attrName]) {
-        this._lockedAttributesValues[attrName] = {};
-      }
-
-      return true;
-    },
-
-    isLocked: function(attrName, options) {
-      const shouldSkipCheck = (options && options.skipcheck);
-      if (!shouldSkipCheck) {
-        const stopProcessing = !this.isLocking(attrName);
-        if (stopProcessing) return;
-      }
-
-      return this.getLockCount(attrName) > 0;
-    },
-
-    getLockCount: function(attrName, options) {
-      const shouldSkipCheck = (options && options.skipcheck);
-      if (!shouldSkipCheck) {
-        const stopProcessing = !this.isLocking(attrName);
-        if (stopProcessing) return;
-      }
-
-      const isGettingValueForSpecificPlugin = options && options.pluginName;
-      if (isGettingValueForSpecificPlugin) {
-        return this._lockedAttributesValues[attrName][options.pluginName] ? 1 : 0;
-      }
-
-      const lockingAttributeValues = Object.values(this._lockedAttributesValues[attrName]);
-      const lockingAttributeValuesSum = lockingAttributeValues.reduce((sum, value) => sum + (value ? 1 : 0), 0);
-
-      return lockingAttributeValuesSum;
-    },
-
-    setLockState: function(attrName, value, options) {
-      const shouldSkipCheck = (options && options.skipcheck);
-      if (!shouldSkipCheck) {
-        const stopProcessing = !this.isLocking(attrName);
-        if (stopProcessing) return this;
-      }
-
-      const isSettingValueForSpecificPlugin = options && options.pluginName;
-      if (!isSettingValueForSpecificPlugin) {
-        console.error('Must supply a pluginName to set a locked attribute lock value');
-        options.pluginName = 'compatibility';
-      }
-
-      if (value) {
-        this._lockedAttributesValues[attrName][options.pluginName] = value;
-      } else {
-        delete this._lockedAttributesValues[attrName][options.pluginName];
-      }
-
-      return this;
+const set = Backbone.Model.prototype.set;
 
+Object.assign(Backbone.Model.prototype, {
+
+  set: function(attrName, attrVal, options = {}) {
```

Comment:
could put these into a LockingModel class

Reply:
This is a bit ambiguous, it's actually a hack to override the set and get functions on a backbone model to implement property locking. I agree it shouldn't live here. I'll have a think.

Reply:
I'll just add a header comment to this file to clarify what it's for if that's ok for the moment?

Reply:
of course! that would be helpful

Reply:
It was actually more self explanatory to add `LockingModel` into the inheritance chain where it's expected rather than try to quantify it in words.

Reply:
Looks far better and more easily read.